invenio_accounts/ext.py

```diff
@@ -27,33 +27,49 @@
 from __future__ import absolute_import, print_function
 
 import pkg_resources
+import redis
+from flask_kvsession import KVSessionExtension
 from flask_security import Security, SQLAlchemyUserDatastore
 from invenio_db import db
+from simplekv.memory.redisstore import RedisStore
+from flask_login import user_logged_in
 
 from .cli import roles as roles_cli
 from .cli import users as users_cli
 from .models import Role, User
+from .sessions import login_listener
 
 
 class InvenioAccounts(object):
     """Invenio-Accounts extension."""
 
-    def __init__(self, app=None):
+    def __init__(self, app=None, sessionstore=None):
         """Extension initialization."""
         self.security = Security()
         self.datastore = None
         if app:
-            self.init_app(app)
+            self.init_app(app, sessionstore=sessionstore)
 
-    def init_app(self, app, use_celery=True):
-        """Flask application initialization."""
+    def init_app(self, app, use_celery=True, sessionstore=None):
+        """Flask application initialization.
+
+        `sessionstore` defaults to `RedisStore(redis.StrictRedis())`
+        """
         self.init_config(app)
 
         # Create user datastore
         self.datastore = SQLAlchemyUserDatastore(db, User, Role)
 
+        # Create session store
+        if sessionstore is None:
+            # TODO: use default from config?
+            sessionstore = RedisStore(redis.StrictRedis())
```

Comment:
Move Redis imports to here, to avoid importing redis if a different sessionstore is provided.